controllers/volumereplicationgroup_controller.go

```diff
@@ -1704,16 +1704,18 @@ func (v *VRGInstance) getStorageClass(namespacedName types.NamespacedName) (*sto
 	return storageClass, nil
 }
 
-func (v *VRGInstance) updateVRGStatus(updateConditions bool) error {
+func (v *VRGInstance) updateVRGStatus(updateAll bool) error {
 	v.log.Info("Updating VRG status")
 
-	if updateConditions {
+	if updateAll {
 		v.updateVRGConditions()
 	}
 
 	v.updateStatusState()
 
-	v.instance.Status.ObservedGeneration = v.instance.Generation
```

Comment:
@ShyamsundarR and @raghavendrabhat the `ObservedGeneration` is updated for every call to `updateVRGStatus`.  I believe this is wrong.  The `ObservedGeneration` should only be updated to match `Generation` if and only if we reached the desired state.  Please correct me if this assumption is wrong.

Reply:
Alright, it is not going to change anything as I have split this PR.  So this change will not make any difference.  But we'll talk about it later. It does not make sense to me to update the generation when nothing has been done because of a failure to read an object for example as in this update: https://github.com/RamenDR/ramen/blob/253e0abae9f713bd3180ebb991a5cdcdd4c2bd9b/controllers/volumereplicationgroup_controller.go#L386